src/plugins/intel_cpu/src/nodes/scatter_update.cpp

```diff
@@ -263,6 +260,462 @@ static std::vector<size_t> getBlockND(const VectorDims& shape) {
     return blockND;
 }
 
+namespace scatter_elements_update {
+
+class ReduceMultiply {
+public:
+    template <typename DT>
+    void operator() (DT* dst_data, const DT* src_data) const {
+        *dst_data *= *src_data;
+    }
+};
+
+class ReduceAdd {
+public:
+    template <typename DT>
+    void operator() (DT* dst_data, const DT* src_data) const {
+        *dst_data += *src_data;
+    }
+};
+
+class ReduceMean {
+public:
+    template <typename DT>
+    void operator() (DT* dst_data, const DT* src_data) const {
+        *dst_data += *src_data;
+    }
+};
+
+class ReduceMaximum {
+public:
+    template <typename DT>
+    void operator() (DT* dst_data, const DT* src_data) const {
+        *dst_data = std::isnan(static_cast<float>(*src_data)) ? *src_data : std::max(*dst_data, *src_data);
+    }
+};
+
+class ReduceMinimum {
+public:
+    template <typename DT>
+    void operator() (DT* dst_data, const DT* src_data) const {
+        *dst_data = std::isnan(static_cast<float>(*src_data)) ? *src_data : std::min(*dst_data, *src_data);
+    }
+};
+
+class ReduceNone {
+public:
+    template <typename DT>
+    void operator() (DT* dst_data, const DT* src_data) const {
+        *dst_data = *src_data;
+    }
+};
+
+template <typename T>
+static T reduction_neutral_value(const ScatterUpdate::Reduction reduction_type) {
+    switch (reduction_type) {
+    case ScatterUpdate::Reduction::MAX:
+        return std::numeric_limits<T>::lowest();
+    case ScatterUpdate::Reduction::MIN:
+        return std::numeric_limits<T>::max();
+    case ScatterUpdate::Reduction::PROD:
+        return T{1};
+    case ScatterUpdate::Reduction::SUM:
+    case ScatterUpdate::Reduction::MEAN:
+    case ScatterUpdate::Reduction::NONE:
+        return T{0};
+    default:
+        OPENVINO_THROW("Neutral value not available for this type of reduction");
+        return 0;
+    }
+}
+
+static ReduceMultiply reduce_multiply;
+static ReduceAdd reduce_add;
+static ReduceMean reduce_mean;
+static ReduceMaximum reduce_maximum;
+static ReduceMinimum reduce_minimum;
+static ReduceNone data_assign;
+
+static inline void getCoordinate(VectorDims& coordinate, size_t offset, const VectorDims& shape) {
+    size_t shapeRank = shape.size();
+    for (int i = shapeRank - 1; i >= 0; i--) {
+        coordinate[i] = offset % shape[i];
+        offset /= shape[i];
+    }
+}
+
+struct TensorIterator {
+    TensorIterator(const VectorDims& squashed_shape, const int64_t squashed_axis) : m_squashed_shape(squashed_shape), m_squashed_axis(squashed_axis) {
```

Comment:
Just for the record, suppose the original tensor is dense, this iterator can be further optimized by squeezing dimensions before `squashed_axis` into 1d, and squeezing dimensions after `squashed_axis` into 1d. and `increment` can be faster.

Reply:
I have a follow up PR to consolidate TensorIterator - https://github.com/openvinotoolkit/openvino/pull/23591/files#diff-29c44414a8e92ddaea4a1e29cd493527e7334b30d12ebd061d4cfe7d65b0edc7 which could make the increment faster and support strided memory at the same time. It still needs time to polish yet, and I am considering include it in our action to extend SEU spec with 1D indices support.